mysys/mf_tempfile.c

```diff
@@ -110,6 +110,29 @@ File create_temp_file(char *to, const char *dir, const char *prefix,
     }
   }
 #elif defined(HAVE_MKSTEMP)
+#if O_TMPFILE
+  static int O_TMPFILE_works= 1;
```

Comment:
the whole block must be in { } otherwise C compiler will complain about definition after statement.